server/actions_projects.go

```diff
@@ -110,21 +115,27 @@ func (s *RegistryServer) ListProjects(ctx context.Context, req *rpc.ListProjects
 		return nil, unavailableError(err)
 	}
 	defer s.releaseStorageClient(client)
+
+	if req.GetPageSize() < 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid page_size: must not be negative")
+	}
+
 	q := client.NewQuery(models.ProjectEntityName)
 	q, err = q.ApplyCursor(req.GetPageToken())
 	if err != nil {
-		return nil, internalError(err)
+		return nil, invalidArgumentError(err)
```

Comment:
Is there any reason an error would occur while applying the cursor for a valid token?

Reply:
Agreed. Cases where the cursor becomes stale seem to be more like a failed precondition or aborted operation anyway. We would want to indicate that the entire sequence should be restarted.